src/main/ruby/core/kernel.rb

```diff
@@ -464,6 +464,14 @@ def tap
     self
   end
 
+  def yield_self
+    if block_given?
+      yield self
+    else
+      Enumerator.new(1) { |yielder| yielder.yield self }
```

Comment:
I wasn't sure if the simplicity of `[self].each` or `[self].to_enum { 1 }` was worth instantiating the Array. Which is preferable?